analyzer/src/main/kotlin/PackageManager.kt

```diff
@@ -54,12 +54,13 @@ typealias ManagedProjectFiles = Map<PackageManagerFactory, List<File>>
 typealias ProjectResults = Map<File, List<ProjectAnalyzerResult>>
 
 /**
- * A class representing a package manager that handles software dependencies. The package manager is referred to by its
- * [managerName]. The analysis of any projects and their dependencies starts in the [analysisRoot] directory using the
+ * A class to represent a package manager of the given [managerName] that handles dependencies for the given
+ * [projectType]. The analysis of any projects and their dependencies starts in the [analysisRoot] directory using the
  * given general [analyzerConfig]. Per-repository configuration is passed in [repoConfig].
  */
 abstract class PackageManager(
     val managerName: String,
+    val projectType: String,
```

Comment:
Is there any manager which may return multiple types? (Not sure about Pub, but maybe this just returns dependencies of different type?)

Reply:
Not right now. I was thinking about this, too, but came to the conclusions that even for managers like Gradle or Bazel (which via plugins support multiple programming languages / ecosystems), you probably still would want to use "Gradle" or "Bazel" here, but different types for the IDs of packages.